File: client/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/rcmd/client/RcmdClientBridge.java
package calebxzhou.rdi.mc.common2.rcmd.client;

import calebxzhou.rdi.mc.rcmd.RcmdSource;

import java.nio.file.Path;

public interface RcmdClientBridge extends RcmdSource {
    Path gameDirectory();

    void executeOnMainThread(Runnable task);

    void setFirmChunkVisible(boolean visible);

    RcmdRecipeSnapshot recipeExportSnapshot();

    RcmdLangSnapshot langExportSnapshot();
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/RcmdClientBridge.kt
package calebxzhou.rdi.mc.rcmd

import calebxzhou.rdi.mc.rcmd.RcmdSource
import java.nio.file.Path

interface RcmdClientBridge : RcmdSource {
    fun gameDirectory(): Path

    fun executeOnMainThread(task: Runnable)

    fun toggleSetFirmSectionsVisible(): Boolean

    fun toggleNowFirmSectionVisible(): Boolean
}
